core/src/main/java/jenkins/model/IdStrategy.java

```diff
@@ -509,7 +513,11 @@ public String idFromFilename(@Nonnull String filename) {
                         } else {
                             break;
                         }
-                        buf.append(Character.valueOf((char)Integer.parseInt(hex.toString(), 16)));
+                        try {
+                            buf.append(Character.valueOf((char)Integer.parseInt(hex.toString(), 16)));
+                        } catch (NumberFormatException x) {
+                            buf.append('$').append(hex);
```

Comment:
Actually this code long predated the security fix. But `CaseInsensitive` is the default for `SecurityRealm`; perhaps few people were using `CaseSensitive` before and thus did not happen to hit this?